core/src/main/java/hudson/model/Node.java

```diff
@@ -406,7 +407,15 @@ public CauseOfBlockage canTake(Queue.BuildableItem item) {
         // Check each NodeProperty to see whether they object to this node
         // taking the task
         for (NodeProperty prop: getNodeProperties()) {
-            CauseOfBlockage c = prop.canTake(item);
+            CauseOfBlockage c;
+            // Commented out to see how the CI fails and it runs successfully when uncommented
+            try {
+                c = prop.canTake(item);
+            } catch (Throwable t) {
+                // We cannot guarantee the task can be taken by this node because something wrong happened
```

Comment:
What happens with the task? does it still remains in the queue?

Reply:
Yes, the task can only be taken by a node that is failing because of a bad implementation of a NodeProperty. This is the wrong behavior reproduced in the test.